frontend/js/src/add-listen/submit-listen-info.tsx

```diff
@@ -0,0 +1,291 @@
+import * as React from "react";
+import { faTimesCircle } from "@fortawesome/free-solid-svg-icons";
+import { throttle as _throttle } from "lodash";
+import DateTimePicker from "react-datetime-picker/dist/entry.nostyle";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { IconProp } from "@fortawesome/fontawesome-svg-core";
+import { faCalendar } from "@fortawesome/free-regular-svg-icons";
+import ListenControl from "../listens/ListenControl";
+import {
+  getAlbumArtFromReleaseMBID,
+  convertDateToUnixTimestamp,
+} from "../utils/utils";
+import GlobalAppContext from "../utils/GlobalAppContext";
+
+export type SubmitListenInfoState = {
+  thumbnailSrc: string;
+  customTimestamp: Boolean;
+  selectedDate: Date;
+  searchField: string;
+  trackResults: Array<ACRMSearchResult>;
+  selectedTrack?: ACRMSearchResult;
+};
+
+export type SubmitListenInfoProps = {
+  onTrackSelect: (trackMetadata: ACRMSearchResult) => void;
+  dateToUnixTimestamp: (event: number) => void;
+  isTrackReset: Boolean;
+  isListenSubmit: Boolean;
+};
+
+export default class SubmitListenInfo extends React.Component<
+  SubmitListenInfoProps,
+  SubmitListenInfoState
+> {
+  static contextType = GlobalAppContext;
+  declare context: React.ContextType<typeof GlobalAppContext>;
+
+  constructor(props: SubmitListenInfoProps) {
+    super(props);
+
+    this.state = {
+      thumbnailSrc: "/static/img/cover-art-placeholder.jpg",
+      customTimestamp: false,
+      selectedDate: new Date(),
+      searchField: "",
+      trackResults: [],
+      selectedTrack: undefined,
+    };
+  }
+
+  componentDidMount() {
+    this.convertToUnix();
+  }
+
+  componentDidUpdate(pp: any, ps: any, ss: any) {
+    const { selectedDate } = this.state;
+    const { isTrackReset, isListenSubmit } = this.props;
+
+    if (ps.thumbnailSrc === "/static/img/cover-art-placeholder.jpg") {
+      this.getCoverArt();
+    }
+    if (ps.selectedDate !== selectedDate) {
+      this.convertToUnix();
+    }
+    if (
+      pp.isTrackReset !== isTrackReset ||
+      pp.isListenSubmit !== isListenSubmit
+    ) {
+      this.removeTrack();
+    }
```

Comment:
This feels like you're trying to solve an issue you shouldn't have with React updates.
I think I understand what the issue might be: the separation of concerns of the components ins't as it should be, and some logic and state in this component here should really be held in the parent instead.
It also seems like isTrackReset and isListenSubmit do the same thing and you could use only isTrackReset.

I think a proper refactoring will be in order, but I suggest we do that as a separate PR after we finish this first implementation.